src/torchmetrics/functional/image/psnrb.py

```diff
@@ -0,0 +1,209 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, Tuple, Union
+
+import torch
+from torch import Tensor, tensor
+from typing_extensions import Literal
+
+from torchmetrics.utilities import rank_zero_warn, reduce
+
+
+def _compute_bef(target: Tensor, dim: Optional[Union[int, Tuple[int, ...]]] = None, block_size=8) -> Tensor:
+    if dim == 3:
+        height, width, channels = target.Size
+    elif dim == 2:
+        height, width = target.Size
+        channels = 1
+    else:
+        raise ValueError("Not a 1-channel/3-channel grayscale image")
```

Comment:
In update method, we define `dim` as:
```
dim: Dimensions to reduce PSNR scores over provided as either an integer or a list of integers.
```
while here is `dim` as an identifier for target image shape, which is likely to cause issues.